develop-docs/sdk/telemetry/logs.mdx

```diff
@@ -0,0 +1,381 @@
+
+// Server-side (Node.js, Bun, Deno) with printf-like syntax
+Sentry.logger.info("Adding item %s for user %s", [itemId, userId], {
+  extra: 123,
+});
+
+Sentry.logger.emit({
+  level: "info",
+  message: "Adding item for user",
+  attributes: {
+    itemId,
+    userId,
+    extra: 123,
+  },
+});
+```
+
+#### Python
+
+```python
+# With f-string like syntax
+Sentry.logger.info('Adding item {item} for user {user}', item=item_id, user=user_id, attributes={ 'extra': 123 });
+
+Sentry.logger.emit(
+  level='info',
+  message='Adding item for user',
+  attributes={
+    'item_id': item_id,
+    'user_id': user_id,
+    extra: 123
+  },
+)
+```
+
+#### PHP
+
+```php
+use function Sentry\logger;
+
+logger()->info('Adding item %s for user %s', [$item_id, $user_id], ['extra' => 123]);
+
+logger()->emit(
+  level: LogLevel::info(),
+  message: 'Adding item for user',
+  attributes: [
+    'item_id': item_id,
+    'user_id': user_id,
+    'extra': 123
+  ],
+);
+```
+
+#### Java
+
+```java
+import io.sentry.Sentry;
+
+// example with MessageFormat based string template
+Sentry.logger().info("Adding item {0,string} for user {1,string}", item_id, user_id);
+
+Sentry.logger().emit(log -> {
+  log.setLevel("info");
+  log.setMessage("Adding item for user");
+  log.setAttribute("item_id", item_id);
+  log.setAttribute("user_id", user_id);
+  log.setAttribute("extra", 123);
+});
+
+// Java - Builder Pattern, where setters return builder for method chaining
+Sentry.Logger.emit(log -> {
+  log.setLevel("info")
+  .setMessage("Adding item for user")
+  .setAttribute("item_id", item_id)
+  .setAttribute("user_id", user_id)
+  .setAttribute("extra", 123)
+});
+
+// Kotlin
+Sentry.Logger.emit("Adding item for user") { // fun emit(block: LogItem.() -> Unit)
+  setLevel("info")
+  setAttribute("item_id", item_id)
+  setAttribute("user_id", user_id)
+  setAttribute("extra", 123)
+}
+```
+
+#### Apple
+
+```swift
+// Swift
+SentrySDK.logger
+  .info(message: "Adding item %s for user %s",
+    params: [itemId, userId],
+    attributes: ["extra": @"123"]
+  )
+
+// Objective-C
+[SentrySDK.logger
+    emit :@"Adding item for user"
+    level: @"info"
+    attributes: @{ @"extra" : @"123", @"item_id" : item_id, @"user_id" : user_id }
+];
+```
+
+#### Threading and Concurrency Considerations
+
+Both the `Sentry.logger.X` and `Sentry.logger.emit` methods are fire-and-forget (have no return value). This means that the SDK can choose to run these methods on a separate thread or queue them up for processing later. This includes evaluating the string template, and running any internal hooks (like `beforeSendLog`). The SDK should ensure that the logs are sent in the order they are received.
+
+## SDK Behavior
+
+### Implementation
+
+If `debug` is set to `true` in SDK init, calls to the Sentry logger API should also print to the console with the appropriate log level. This will help debugging logging setups.
```

Comment:
Does the Sentry logger API print all logs? If yes, maybe we need an option to control which log levels the logger should log. Also OK to fix it in the future and skip it for now.

Reply:
It prints all logs. I agree we probably need another option, will mark it as a TODO.